.github/workflows/rc.yml

```diff
@@ -0,0 +1,127 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: RC
+on:
+  push:
+    branches:
+      - '**'
+      - '!dependabot/**'
+    tags:
+      - '*-rc*'
+  pull_request:
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+permissions:
+  contents: read
+
+jobs:
+  archive:
+    name: Archive
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Prepare for tag
+        if: github.ref_type == 'tag'
+        run: |
+          version=${GITHUB_REF_NAME%-rc}
+          version=${version#v}
+          rc=${GITHUB_REF_NAME#*-rc}
```

Comment:
I think we should make the RC number configurable in the GH action

Reply:
The RC number is configurable: it is taken from the tag which kicks off the GH action run. 

i.e. we kick off the action by creating a tag `v0.1.0-rc1` and starting the action:

```console
version=${GITHUB_REF_NAME%-rc}
version=${version#v}
```
After this, `version` == `0.1.0-rc1`

```console
rc=${GITHUB_REF_NAME#*-rc}
```

After this, `rc` == `1`.

Thus the RC is configured by the tag that triggers the GH action.